#1235 binarytype, encoding GHP update (#1388)
AbsaOSS/enceladus: _docs/2.0.0/usage/schema.md

| [grouping_separator](#grouping_separator) | any numeric type           | Character to mark boundaries between orders of magnitude, usually to mark thousands, millions etc.                                                    | *\_*            | *,*                                      |
| [minus_sign](#minus_sign)                 | any numeric type           | Character to mark the number is negative.                                                                                                             | *N*             | *-*                                      |
| [allow_infinity](#allow_infinity)         | float & double             | Flag indicating if the column accepts infinity as a value (and positive/negative numbers which are too large are converted to *infinity*/*-infinity*) | *true*          | *false*                                  |
| [radix](#radix)                           | long, integer, short, byte | The base of the numbers provided                                                                                                                      | *hex*           | *10*                                     |

> | [encoding](#encoding)                     | enumeration(base64, none)  | Encoding used for string->binary conversion                                                                                                           | *base64*,*none* | (no encoding present)                    |

Reviewer: what is `enumeration(base64, none)`

Author: that was supposed to mean: only these two values are allowed

Author: re-expressed, hope it is clearer :)

Reviewer: I commented on the wrong thread. Sorry.

> | [encoding](#encoding)                     | string ("base64" or "none") | Encoding used for string->binary conversion                                                                                                           | *base64*,*none* | (no encoding present)                    |

Reviewer: I would say that the correct line would be this.
- It is simply on a binary. Description says more
- It is for a string to binary conversions.
- And the default is none (or not used), but not used does not have to be there. Then we would have to say it everywhere. That doesn't make sense

```suggestion
| [encoding](#encoding)                     | binary                      | Encoding is used for string to binary conversion                                                                                                      | *base64*,*none* | *none*                                   |
```

Author: ok

| [minus_sign](#minus_sign)                 | any numeric type            | Character to mark the number is negative.                                                                                                             | *N*             | *-*                                      |
| [allow_infinity](#allow_infinity)         | float & double              | Flag indicating if the column accepts infinity as a value (and positive/negative numbers which are too large are converted to *infinity*/*-infinity*) | *true*          | *false*                                  |
| [radix](#radix)                           | long, integer, short, byte  | The base of the numbers provided                                                                                                                      | *hex*           | *10*                                     |

> | [encoding](#encoding)                     | binary                      | Encoding is used for string to binary conversion                                                                                                      | *base64*,*none* | *none*                                   |

Reviewer 2: *none* is strictly speaking not default here, only implied resulting in possible warnings. 🤔

Author: Yes. I was about to point that out, but then I realized we can hardly clarify it completely in the table row.

The description of the `encoding` metadata key-value explains this in sufficient detail, so I was hoping it should suffice https://github.com/AbsaOSS/enceladus/pull/1388/files#diff-e4c9947563dc83be794d2f43bd711113R397

I'll just make a link to it from here.

If `radix` is specified as anything other than the default 10, [pattern](#pattern)
value will be ignored.

### encoding

<sup>**Supported by types:** [Binary](#Binary)</sup>

**When a string value is being converted to binary**, the supplied `encoding` indicates how the values are going to be 
treated. This applies for the default value, too:
 - `none` - the input will get cast as-is to binary. E.g. "abc" -> [97, 98, 99]
 - `base64` - the input is considered as Base64-encoded and will get unencoded. Contrary to the basic Spark behavior of
 `unbase64` (which skips characters invalid for Base64), this will result in an error.

> If encoding is missing altogether would it be needed (e.g. default value is given), `ValidationWarning` is issued and

Reviewer: I am not sure what this sentence is saying

Author: True, rephrased.

 the encoding value is considered to be `none`.
 

> `encoding` is not considered if BinaryType is already found in the input (so conversion is happening there).

Reviewer: no conversion?

Author: Yes, thanks.
